src/stores/service/store.service.ts

```diff
@@ -1,104 +1,114 @@
 import { HttpStatus, Injectable } from '@nestjs/common';
 import { InjectModel } from '@nestjs/mongoose';
 import { Store } from '../schema/store.schema';
-import { Model } from 'mongoose';
+import { Model, Types } from 'mongoose';
 import { CreateStoreDTO } from '../dto/create-store.dto';
 import { UpdateStoreDTO } from '../dto/update-store.dto';
 import { AppError } from '~/common/app-error.common';
 
 @Injectable()
 export class StoreService {
-  constructor(
-    @InjectModel(Store.name)
-    private readonly storeModel: Model<Store>,
-  ) {}
-
-  async createStore(createStoreDTO: CreateStoreDTO) {
-    const newStore = new this.storeModel(createStoreDTO);
-
-    await newStore.save();
-
-    return {
-      message: 'Store created successfully',
-    };
-  }
-
-  /**
-   * only for super-admin
-   */
-  async getAllStores() {
-    const stores = await this.storeModel.find().lean().exec();
-
-    return stores.map((store) => ({
-      ...store,
-      _id: store._id.toString(),
-    }));
-  }
-
-  /**
-   * allow a retailer get their store details
-   */
-  async getMyStore(id: string) {
-    const stores = await this.storeModel.find({
-      owner: '66bf4d123627d4e6dff790c5',
-    });
-
-    if (!stores) {
-      throw new AppError(
-        `No stores found for user with ID ${id}`,
-        HttpStatus.NOT_FOUND,
-      );
+    constructor(
+        @InjectModel(Store.name)
+        private readonly storeModel: Model<Store>,
+    ) { }
+
+
+    async createStore(ownerId: string, createStoreDto: CreateStoreDTO) {
+        const storeCount = await this.storeModel.countDocuments({ owner: ownerId });
+
+        if (storeCount >= 10) {
+            throw new AppError(
+                'You cannot own more than 10 stores',
+                HttpStatus.BAD_REQUEST
+            );
+        }
```

Comment:
You can try to move this to your schema to check if it'll work. 
In the `@Prop()` you'll pass this into it as a property. 

```javascript
validate: {
      validator: function (value) {
        // Check if the array length is less than or equal to 10
        return value.length <= 10;
      },
      message: 'The items array cannot contain more than 10 elements.'
    }
```
This will only run on `save` and `create`. but you must call the

Reply:
the relationship between the user and store is unidirectional , only the store is aware of the relationship

Reply:
doesn't affect the validation